packages/controls/src/serialization/function-serialization.ts

```diff
@@ -6,7 +6,16 @@ export function isFunction(value: any): value is AnyFunction {
   return typeof value === 'function'
 }
 
+type ResolveCallPromise<T extends AnyFunction> = (
+  value: SerializedFunctionReturnType<T>,
+) => void
```

Comment:
Simplified version of https://github.com/makeswift/makeswift/pull/864/files#diff-27828807c3388a8fc03da7ddb6fb171dc73e668a46695517bc7cd651c3cd017bL52